zones.go

```diff
@@ -30,6 +33,51 @@ type Zone struct {
 	Tags                  []ResourceTag     `json:"tags,omitempty"`
 }
 
+// List fetches all the zones
+func (zone *Zone) List(ctx context.Context, client *Client) (<-chan interface{}, <-chan error) {
+	outChan := make(chan interface{}, client.PageSize)
+	errChan := make(chan error, 1)
+
+	go func() {
+		defer close(outChan)
+		defer close(errChan)
+
+		page := 0
+		count := 0
+
+		req := &ListZones{
+			DomainID: zone.DomainID,
+			ID:       zone.ID,
+			Name:     zone.Name,
+			PageSize: client.PageSize,
+		}
+
+		for {
+			req.Page = page
+
+			resp, err := client.RequestWithContext(ctx, req)
+			if err != nil {
+				errChan <- err
+				break
+			}
+
+			zones := resp.(*ListZonesResponse)
+			for _, zone := range zones.Zone {
+				count++
+				outChan <- zone
+			}
+
+			if count >= zones.Count {
+				break
```

Comment:
Risky! Between pagination call the count can increase or decrease. You're best bet is to stop paginating when the page result is empty.

Reply:
don't trust `Count` and paginate until a failure occurs?

Reply:
if the result size is smaller than the page size we abort.